Kinetic_space_partition/include/CGAL/KSP_3/Intersection_graph.h

```diff
@@ -0,0 +1,445 @@
+// Copyright (c) 2023 GeometryFactory Sarl (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Sven Oesau, Florent Lafarge, Dmitry Anisimov, Simon Giraudot
+
+#ifndef CGAL_KSP_3_INTERSECTION_GRAPH_H
+#define CGAL_KSP_3_INTERSECTION_GRAPH_H
+
+#include <CGAL/license/Kinetic_space_partition.h>
+
+// Boost includes.
+#include <boost/graph/adjacency_list.hpp>
+
+// CGAL includes.
+#include <CGAL/Cartesian_converter.h>
+#include <CGAL/Polygon_2.h>
+
+// Internal includes.
+#include <CGAL/KSP/utils.h>
+
+namespace CGAL {
+namespace KSP_3 {
+namespace internal {
+
+#ifdef DOXYGEN_RUNNING
+#else
+
+template<typename GeomTraits, typename IntersectionKernel>
+class Intersection_graph {
+
+public:
+  using Kernel = GeomTraits;
+  using Intersection_kernel = IntersectionKernel;
+
+  using Point_2 = typename Intersection_kernel::Point_2;
+  using Point_3 = typename Intersection_kernel::Point_3;
+  using Segment_3 = typename Intersection_kernel::Segment_3;
+  using Line_3 = typename Intersection_kernel::Line_3;
+  using Polygon_2 = typename CGAL::Polygon_2<Intersection_kernel>;
+
+  using Inexact_FT = typename Kernel::FT;
+
+  struct Vertex_property {
+    Point_3 point;
+    Vertex_property() {}
+    Vertex_property(const Point_3& point) : point(point) {}
+  };
+
+  using Kinetic_interval = std::vector<std::pair<Inexact_FT, Inexact_FT> >;
+
+  struct Edge_property {
+    std::size_t line;
+    std::size_t order;
+    std::map<std::size_t, std::pair<std::size_t, std::size_t> > faces; // For each intersecting support plane there is one pair of adjacent faces (or less if the edge is on the bbox)
+    std::set<std::size_t> planes;
+    std::set<std::size_t> crossed;
+    std::map<std::size_t, Kinetic_interval> intervals; // Maps support plane index to the kinetic interval. std::pair<FT, FT> is the barycentric coordinate and intersection time.
+    Edge_property() : line(std::size_t(-1)), order(edge_counter++) { }
+
+    Edge_property(const Edge_property& other) {
```

Comment:
@lrineau:  Why do we get this warning:
````
/Users/magritte/cgal_root/CGAL-6.0-Ic-209/include/CGAL/KSP_3/Intersection_graph.h:67:26: warning: definition of implicit copy constructor for 'Edge_property' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
````

Reply:
Taken the way it is implemented we might write `= default`  but I simply do not understand the warning.
And if we kept it, there should be an initializer list @soesau

Reply:
Tu commentes le code qui a déjà été corrigé par Sven:
https://github.com/soesau/cgal/commit/b090427bad6e240363c81c6cf4c026ccb01cc096#diff-afbbbcb5e5b425571047333fd2b13928db303c82022ec58e107ee461ccff44ae

Il a ajouté le copy-constructor.

Reply:
Thanks. Do you agree that I should replace it with `= default` ?

Reply:
Both copy-constructor and copy-assignment can be implemented with `=default`. Or even removed: the compiler will generate them, as well as the move-constructor and move-assignment.